src/Farmer/Arm/Compute.fs

```diff
@@ -75,7 +76,13 @@ type VirtualMachine =
                      {| computerName = this.Name.Value
                         adminUsername = this.Credentials.Username
                         adminPassword = this.Credentials.Password.ArmExpression.Eval()
-                        customData = this.CustomData |> Option.map (System.Text.Encoding.UTF8.GetBytes >> Convert.ToBase64String) |> Option.toObj |}
+                        customData = this.CustomData |> Option.map (System.Text.Encoding.UTF8.GetBytes >> Convert.ToBase64String) |> Option.toObj 
+                        linuxConfiguration = 
+                            match this.LinuxConfiguration with
+                            | Some linuxConfiguration -> 
+                                {| disablePasswordAuthentication = linuxConfiguration.DisablePasswordAuthentication
+                                   ssh = {| publicKeys = linuxConfiguration.PublicKeys |> List.map (fun k -> {| path = k.path;keyData = k.keyData |}) |} |}
+                            | None -> Unchecked.defaultof<_> |}
```

Comment:
Check on some of the other ArmResource but I'm pretty sure most dont use  Unchecked.defaultof<_>.

Reply:
done